packages/services/guard/src/index.ts

```diff
@@ -1,2 +1,10 @@
-// export { Guard } from './guard.gen'
-// export * from './signer'
+import { Address, Hex, Bytes, Signature } from 'ox'
+
+export interface GuardSigner {
+  readonly address: Address.Address
+
+  sign(wallet: Address.Address, chainId: number, digest: Bytes.Bytes, message: Hex.Hex): Promise<Signature.Signature>
```

Comment:
Can `digest` be derived from the `message`? Do we need both here?

Reply:
The reason we need both is because the way `digest` is derived from `message` depends on what the `message` is, which the guard service cannot know at this point. E.g. when it's TypedData, the message is the JSON but the `digest` is derived from the encoded TypedData. Similarly with attestation (message is the JSON but digest derived from the encoded bytes.)

Reply:
I don't want to move that logic here (and dependency on wallet-primitives) as this should be a dumb service

Reply:
Is the guard signing the digest? 
I'd argue it doesn't make sense to include the message for validation if the result it's signing can't be derived. Because a malicious actor could send a valid payload with a malicious digest and guard wouldn't be able to tell.